crates/net/network/src/transactions/config.rs

```diff
@@ -248,6 +248,54 @@ where
     }
 }
 
+/// Sharded mempool announcement filtering policy
+#[derive(Debug, Clone)]
+pub struct ShardedMempoolAnnouncementFilter {
+    pub shard_bits: u8,
+    pub node_id: Option<B256>,
+}
+
+impl Default for ShardedMempoolAnnouncementFilter {
+    fn default() -> Self {
+        Self {
+            shard_bits: 4,
+            node_id: None,
+        }
+    }
+}
+
+impl ShardedMempoolAnnouncementFilter {
+    pub fn new_with_node_id(node_id: B256) -> Self {
+        Self {
+            shard_bits: 4,
+            node_id: Some(node_id),
+        }
+    }
+}
+
+impl AnnouncementFilteringPolicy for ShardedMempoolAnnouncementFilter {
```

Comment:
since this is layered on behaviour, we can make this generic over another policy type that we forward the non type 3 txs to

like `SharedMempoolAnnounceFilter<T>`